frontend/frontend.go

```diff
@@ -432,6 +473,31 @@ func (f *Frontend) ArmSubscriptionAction(writer http.ResponseWriter, request *ht
 	subId := request.PathValue(PathSegmentSubscriptionID)
 	f.cache.SetSubscription(subId, &subscription)
 
+	var doc *SubscriptionDocument
+	doc, err = f.dbClient.GetSubscriptionDoc(ctx, subId)
+	if err != nil && !errors.Is(err, ErrNotFound) {
```

Comment:
This error handling logic flow seems a little clunky to me.  I get that you're trying to handle actual error conditions first, but I think it would read better if you structured it as you've done elsewhere:

```
if err != nil {
        if errors.Is(err, ErrNotFound) {
                // initialize new SubscriptionDocument
        } else {
                // write error and return
        }
} else {
        // use found SubscriptionDocument
}
```

Just my opinion though.  If you disagree then disregard.

Reply:
Don't have strong opinions here, your ask makes sense, happy to add the change while im making the other one.

Reply:
I remember what my thinking was here. Since this call is being used to create or update, i was trying to avoid setting the `doc.Subscription` value before checking for all possible errors. I think what you've suggested does the same so that works for me, i think i just missed this way of getting to the same end. Thanks!